components/suggest/src/store.rs

```diff
@@ -3916,6 +3949,22 @@ pub(crate) mod tests {
             },],
         );
 
+        // Make sure the bbb suggestion hasn't been dismissed even though it
+        // has the same key as the first aaa suggestion.
+        assert_eq!(
+            suggestions_1[0].dismissal_key(),
+            suggestions_0[0].dismissal_key()
+        );
+        assert_eq!(
+            store.fetch_suggestions(SuggestionQuery::dynamic("bbb", &["bbb"])),
+            vec![Suggestion::Dynamic {
+                suggestion_type: "bbb".to_string(),
+                data: None,
+                dismissal_key: Some("dk0".to_string()),
+                score: DEFAULT_SUGGESTION_SCORE,
+            },],
+        );
```

Comment:
The opposite here please, i.e., `store.inner.is_dismissed_by_suggestion` returns false for the bbb suggestion